tools/util/getDiff.ts

```diff
@@ -0,0 +1,43 @@
+import { execSync } from "child_process";
+import { basename, dirname } from "path";
+
+export type ValidEventName = "push" | "pull_request" | "uncommitted";
+
+/**
+ * Gets an array of all the changed folders using the git diff
+ * @returns {Promise<string[]>} Array of unique paths to the changed folders
+ */
+export default function getDiff(
+	type: "addedModified" | "removed" | "all" = "addedModified"
+): string[] {
+	const commands: Record<ValidEventName, string> = {
+			push: "HEAD HEAD^",
+			pull_request: `origin/main HEAD`, //`HEAD ${process.argv[3] ?? "main"}`,
```

Comment:
i dont think this works, especially as CI didnt pass

Reply:
You said you would prove to me that this works by adding a `console.log()` for each presence to show that it passed Presence Validator

Reply:
Added